expression/builtin_cast_vec.go

```diff
@@ -118,3 +118,34 @@ func (b *builtinCastIntAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.
 func (b *builtinCastIntAsRealSig) vectorized() bool {
 	return true
 }
+
+func (b *builtinCastTimeAsDurationSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	arg0, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(arg0)
+	if err := b.args[0].VecEvalTime(b.ctx, input, arg0); err != nil {
+		return err
+	}
+	arg0s := arg0.Times()
+	result.ResizeGoDuration(n, false)
+	result.MergeNulls(arg0)
+	ds := result.GoDurations()
+	for i, t := range arg0s {
+		if result.IsNull(i) {
+			continue
+		}
+		d, err := t.ConvertToDuration()
+		if err != nil {
+			return err
+		}
+		ds[i] = d.Duration
```

Comment:
Do we need this line ```ds[i], err = d.RoundFrac(int8(b.tp.Decimal))``` like https://github.com/pingcap/tidb/blob/master/expression/builtin_cast.go#L1361

Reply:
Fixed, and added a related test.